Set4/30/calculator/getOperator.cc

```diff
@@ -0,0 +1,14 @@
+#include "calculator.ih"
+
+    // by expression.cc
+
+bool Calculator::getOperator()
+{
+    d_op = d_parser.next()[0];                      // parse operator symbol
+    
+    string const validOperators = "+-*/%";          // verify
+    if (validOperators.find(d_op) == string::npos)
+        return false;
+    
+    return true;
```

Comment:
```suggestion
   d_op = d_parser.next();
   return (d_op == "+" || d_op == "-" || d_op == "*" || d_op == "/" || d_op == "%"))
```
The parsed string might be empty if the input is EOLN, so d_parser.next()[0] may cause end of line error/segfault. The exercise is vague about what "malformed" inputs are so it's good practice to be more defensive here. 

My suggestion parses the next element and checks if this is one of the accepted operators.

Reply:
This gives me an error since the string isn't converted to a character

Reply:
That adds complications to the switch. I think I will keep this as is for now?

Reply:
I am not getting issues if I enter something like "5 [enter]" at least!

Reply:
Alright! After I send over set 3 I will implement this!